src/app/forms/custom-sections-v2/general-vehicle-details/general-vehicle-details.component.ts

```diff
@@ -257,59 +310,55 @@ export class GeneralVehicleDetailsComponent extends EditBaseComponent implements
-		const vehicleConfigurationControl = this.form.get('techRecord_vehicleConfiguration');
-		const vehicleConfigurationValue = vehicleConfigurationControl?.getRawValue() as VehicleConfiguration;
-
-		if (!vehicleConfigurationValue) {
-			return;
-		}
+		const vehicleConfigurationValue = this.form
+			.get('techRecord_vehicleConfiguration')
+			?.getRawValue() as VehicleConfiguration;
 
-		if (vehicleType === VehicleTypes.HGV) {
-			if (vehicleConfigurationValue === null) {
-				this.bodyTypes = [];
-			}
+		if (!vehicleConfigurationValue) return;
 
-			// When vehicle configuration is set to articulated, update the body type description and code
-			if (vehicleConfigurationValue === VehicleConfiguration.ARTICULATED) {
-				this.bodyTypes = getOptionsFromEnum(Array.from(articulatedHgvBodyTypeCodeMap.values()).flat());
-
-				this.form.patchValue({
-					techRecord_bodyType_description: BodyTypeDescription.ARTICULATED,
-					techRecord_bodyType_code: BodyTypeCode.A,
-				});
-			}
-
-			// When vehicle configuration is rigid, clear artic body description and code
-			if (vehicleConfigurationValue === VehicleConfiguration.RIGID) {
-				this.bodyTypes = getOptionsFromEnum(Array.from(hgvBodyTypeCodeMap.values()).flat());
-
-				const bodyTypeCode = this.form.get('techRecord_bodyType_code')?.getRawValue();
-				const bodyTypeDescription = this.form.get('techRecord_bodyType_description')?.getRawValue();
-
-				if (bodyTypeCode === BodyTypeCode.A || bodyTypeDescription === BodyTypeDescription.ARTICULATED) {
+		// Set bodyTypes based on vehicle type and configuration
+		switch (vehicleType) {
+			case VehicleTypes.HGV:
+				if (vehicleConfigurationValue === VehicleConfiguration.ARTICULATED) {
+					this.bodyTypes = getOptionsFromEnum([...articulatedHgvBodyTypeCodeMap.values()].flat());
 					this.form.patchValue({
-						techRecord_bodyType_description: null,
-						techRecord_bodyType_code: null,
+						techRecord_bodyType_description: BodyTypeDescription.ARTICULATED,
+						techRecord_bodyType_code: BodyTypeCode.A,
 					});
+				} else if (vehicleConfigurationValue === VehicleConfiguration.RIGID) {
+					this.bodyTypes = getOptionsFromEnum([...hgvBodyTypeCodeMap.values()].flat());
+					const code = this.form.get('techRecord_bodyType_code')?.getRawValue();
+					const desc = this.form.get('techRecord_bodyType_description')?.getRawValue();
+					if (code === BodyTypeCode.A || desc === BodyTypeDescription.ARTICULATED) {
+						this.form.patchValue({
+							techRecord_bodyType_description: null,
+							techRecord_bodyType_code: null,
+						});
+					}
+				} else if (vehicleConfigurationValue === null) {
+					this.bodyTypes = [];
 				}
-			}
-		} else {
-			const options = vehicleBodyTypeDescriptionMap.get(vehicleType)?.values() || [];
-			this.bodyTypes = getOptionsFromEnum(Array.from(options).flat());
+				break;
+
+			case VehicleTypes.TRL:
+				this.bodyTypes = getOptionsFromEnum([...hgvBodyTypeCodeMap.values()].flat());
```

Comment:
Body types for HGV and TRL are different:

```suggestion
				this.bodyTypes = getOptionsFromEnum([...trlBodyTypeCodeMap.values()].flat());
```

I'm also not sure if this is the correct approach as for TRLs the body type list should not depend on vehicle configuration and should be populated at the start regardless of what is selected. It currently appears blank, and does not have the same behaviour as develop.